mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -270,17 +263,28 @@ impl PlatformCoinWithTokensActivationOps for TendermintCoin {
             TendermintActivationPolicy::with_private_key_policy(tendermint_private_key_policy)
         };
 
-        TendermintCoin::init(
+        let coin = TendermintCoin::init(
             &ctx,
-            ticker,
+            ticker.clone(),
             conf,
             protocol_conf,
             activation_request.nodes,
             activation_request.tx_history,
             activation_policy,
             is_keplr_from_ledger,
         )
-        .await
+        .await?;
+
+        if let TendermintActivationPolicy::PublicKey(_) = coin.activation_policy {
+            if !ctx.disable_watchers_globally() && coin.is_supported_by_watchers() {
+                return MmError::err(TendermintInitError {
+                    ticker,
+                    kind: TendermintInitErrorKind::CantUseWatchersWithPubkeyPolicy,
+                });
+            }
+        }
```

Comment:
Why not do the opposite, if watchers can never be implemented for pubkey policy then we should implement `is_supported_by_watchers` for Tendermint as
```rust
fn is_supported_by_watchers(&self) -> bool { !matches!(self.activation_policy, TendermintActivationPolicy::PublicKey(_)) && false }
```
We later remove the false when we implement watchers for tendermint. We wouldn't need this code block / error at all as when we enable pubkey policy, watchers will not be used. Otherwise when we implement watchers for tendermint this will always return an error again if we forgot about fixing it.

Reply:
This
```rust
fn is_supported_by_watchers(&self) -> bool { !matches!(self.activation_policy, TendermintActivationPolicy::PublicKey(_)) && false }
```
will cause a clippy error, I am fine with keeping the error for now if you don't want to deny clippy.

Reply:
Please resolve this if you wanna keep the current code

Reply:
Are we ever going to allow users to use watcher nodes with pubkey-online mode? I'm not sure how that would work during swaps. It will be super complicated thing to implement for sure (e.g., connecting user's hardware wallet to remote KDF instance).

Reply:
>  (e.g., connecting user's hardware wallet to remote KDF instance).

Remote KDF instance will be always online so it won't need watchers. For external wallets (e.g. Keplr), we will need watchers for sure and since most external wallets let you sign a transaction, watchers should work.

Reply:
Resolved in Element DMs.